### PR TITLE
Fix handling of R mode

### PR DESCRIPTION
**Pull request by fviale**
- wrong code mirror mode was defined
 - better usage of lowercase

### CODE REVIEW

File: app/scripts/proactive/view/dom.js [hunk 1]
```diff
@@ -745,14 +745,15 @@ define(
...
-                var selectedLanguage = languageElement.options[languageElement.selectedIndex].value.toLowerCase();
+                var selectedLanguage = languageElement.options[languageElement.selectedIndex].value;
                 if (isUrl && (!selectedLanguage || selectedLanguage == '')) {
                     var indexExt = inputValue.lastIndexOf('.');
                     if (indexExt > -1) {
                         var extension = inputValue.substring(indexExt+1, inputValue.length);
                         selectedLanguage = config.extensions_to_languages[extension.toLowerCase()] || '';
                     }
                 }
+                selectedLanguage = selectedLanguage.toLowerCase()
```
[the remark this answers is not in the document]
Comment (fviale) [reply app/scripts/proactive/view/dom.js]: I simply lowercase "after" it has been set, without changing the semantic.

Before my change, these instructions could lead to an error, as toLowerCase() was used before the test:

var selectedLanguage = languageElement.options[languageElement.selectedIndex].value.toLowerCase();

if (isUrl && (!selectedLanguage || selectedLanguage == '')) {
}

[the remark this answers is not in the document]
Comment (chloegugli) [reply app/scripts/proactive/view/dom.js]: There is already a toLowerCase line 753, You'll do it twice if you enter the if. 
You could just add it to the line 748

[the remark this answers is not in the document]
Comment (fviale) [reply app/scripts/proactive/view/dom.js]: this toLowerCase() is on extension, not on selectedLanguage

[the remark this answers is not in the document]
Comment (chloegugli) [reply app/scripts/proactive/view/dom.js]: Approved

**Pull request merged**